index.js

```diff
@@ -1,47 +1,89 @@
 'use strict';
 
-//setup requirements of express server
+//--SETUP AND INITIALIZATION
+//Setup requirements of express server
 require('dotenv').config();
 const cors = require('cors');
 const express = require('express');
 const app = express();
-
-//enable cors
+const axios = require('axios');
+//Enable cors
 app.use(cors());
-
-//define PORT based on environment
+//Define PORT based on environment
 const PORT = process.env.PORT;
 
-//server paths
+
+//--SERVER PATHS
 app.get('/hello', (req, res) => res.send('Hi there! :)'));
 app.get('/weather', serveWeather);
-app.get('/*', (req,res) => res.status(500).json({error: 'Server resource does not exist!'}));
+app.get('/movies', serveMovie);
+app.get('/*', (req, res) => res.status(500).json({ error: 'Server resource does not exist!' }));
+
 
-//server Data
-let weatherData = require('./weather.json');
+//--SERVER DATA HANDLING
+//Retrieve Weather data from weatherbit.io API
+async function retrieveWeather(lat, lon) {
+  return await axios.get(`https://api.weatherbit.io/v2.0/forecast/daily?key=${process.env.WEATHER_API_KEY}&lat=${lat}&lon=${lon}&days=7&units=I`);
+}
+
+//Retrieve Movie data from TMDB API
+async function retrieveMovies(place) {
+  return await axios.get(`https://api.themoviedb.org/3/search/movie?api_key=${process.env.MOVIE_API_KEY}&language=en-US&query=${place}&page=1&include_adult=false`);
+}
 
 //Return classes
 class Forecast {
-  constructor(date, description){
-    this.date = date;
-    this.description = description;
+  constructor(obj) {
+    this.date = obj.datetime;
+    this.lowTemp = obj.low_temp;
+    this.highTemp = obj.high_temp;
+    this.description = obj.weather.description;
+    this.icon = obj.weather.icon;
   }
 }
 
-//server actions
-function serveWeather(req, res) {
-  let search = req.query;
+class Movie {
+  constructor(obj) {
+    this.title = obj.title;
+    this.description = obj.overview;
+    this.posterUrl = `https://image.tmdb.org/t/p/w342${obj.poster_path}`;
+    this.release = obj.release_date;
+    this.averageVotes = obj.vote_average;
+    this.totalVotes = obj.vote_count;
+    this.popularity = obj.popularity;
+  }
+}
+
+//--SERVER ACTIONS
+//Handle weather information query
+async function serveWeather(req, res) {
+  const { lat, lon } = req.query;
+
+  let weatherResponse = await retrieveWeather(lat, lon);
+  let weatherData = weatherResponse.data.data;
 
-  let weatherResult = weatherData.find(place => Math.floor(parseFloat(search.lat)) === Math.floor(parseFloat(place.lat)) && Math.floor(parseFloat(search.lon)) === Math.floor(parseFloat(place.lon)));
+  let forecastArray = weatherData ? weatherData.map(day => new Forecast(day)) : false;
+
+  if (forecastArray) {
+    res.send(forecastArray); // send back forecast array
+  } else {
+    res.status(404).json({ error: 'Weather data not found' }); //return not found error
+  }
+}
```

Comment:
Great work on the lab!

I notice you are using try catch blocks for movies but not here for `serveWeather`. We don't always have to use try catch blocks of course, but do you expect a failure to occur anywhere here?

Reply:
I think that might be oversight on my part. I'm finding that generally my errors come from the Heroku API itself not replying to my front end. The only error I'd really expect from weatherbit or TMDB is being down for maintenance, but I should still handle that accordingly. As I refactor today I plan to reevaluate my error handling, so I'll look at getting this setup to handle those cases today. Good catch :)

Reply:
Sounds good! Yes, I think I overuse try catch block haha (not good). Anyway just a question here so I see into your thoughts

![pinky](https://media.giphy.com/media/sytgHDEeDuGWc/giphy.gif)